website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -409,13 +417,23 @@ Specify using ids:
 
 ## spec.role
 
-`Role` is a required field and is necessary to tell Karpenter which identity nodes from this `EC2NodeClass` should assume. If using the [Karpenter Getting Started Guide]({{<ref "../getting-started/getting-started-with-karpenter" >}}) to deploy Karpenter, you can use the `KarpenterNodeRole-$CLUSTER_NAME` role provisioned by that process.
+`Role` is an optional field and is necessary to tell Karpenter which identity nodes from this `EC2NodeClass` should assume. You must specify one of `role` or `instanceProfile` when creating a Karpenter `EC2NodeClass`. If using the [Karpenter Getting Started Guide]({{<ref "../getting-started/getting-started-with-karpenter" >}}) to deploy Karpenter, you can use the `KarpenterNodeRole-$CLUSTER_NAME` role provisioned by that process.
 
 ```yaml
 spec:
   role: "KarpenterNodeRole-$CLUSTER_NAME"
 ```
 
+## spec.instanceProfile
+
+`InstanceProfile` is an optional field and is neccesary to tell Karpenter which identity nodes from this `EC2NodeClass` should assume. You must specify one of `role` or `instanceProfile` when creating a Karpenter `EC2NodeClasss`. If you use the `instanceProfile` field instead of `role`, Karpenter will not manage the InstanceProfile on your behalf.
```

Comment:
How I'd write it:
```suggestion
You must specify one of `role` or `instanceProfile` for a Karpenter EC2NodeClass. `instanceProfile` is an optional field that you can use to tell Karpenter the AWS IAM identity that nodes from this EC2NodeClass should assume.  If you use the `instanceProfile` field instead of `role`, Karpenter does not manage the instance profile on your behalf.

If you have an existing EC2NodeClass that defines `.spec.instanceProfile`, you cannot switch that EC2NodeClass
to use `.spec.role`; instead, if you need to, you can make a new EC2NodeClass and delete the old one.
```